{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "3a1355a3",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Allowing multi-threading with up to 48 threads.\n"
     ]
    }
   ],
   "source": [
    "setwd(\"/mnt/lareaulab/reliscu/projects/NSF_GRFP/analyses/bulk/hahn_2023/cortex\")\n",
    "\n",
    "library(dplyr)\n",
    "library(data.table)\n",
    "\n",
    "source(\"/mnt/lareaulab/reliscu/code/FindModules/FindModules.R\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ef11b82",
   "metadata": {},
   "source": [
    "Here I run FM to (hopefully) find modules representing each of the cell types present in the single-cell data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b30fb730",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 52633     3\n"
     ]
    }
   ],
   "source": [
    "expr_tpm <- fread(\"hahn_2023_cortex_STAR_TPM_SampleNetworks/All_06-23-55/hahn_2023_cortex_STAR_TPM_All_68_outliers_removed.csv\", data.table=FALSE)\n",
    "\n",
    "# For duplicate genes, choose row with highest mean expression\n",
    "\n",
    "mean_expr <- data.frame(\n",
    "    Index=1:nrow(expr_tpm), \n",
    "    Gene=expr_tpm[,1], \n",
    "    Expr=rowMeans(expr_tpm[,-1])\n",
    ")\n",
    "\n",
    "mean_expr <- mean_expr %>%\n",
    "    group_by(Gene) %>%\n",
    "    slice_max(Expr)\n",
    "\n",
    "print(dim(mean_expr))\n",
    "\n",
    "expr_tpm <- expr_tpm[mean_expr$Index,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fab5fc01",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 21053\n"
     ]
    }
   ],
   "source": [
    "# Subset to genes in the top X percentile of mean expression\n",
    "\n",
    "prob <- .6\n",
    "mean_expr <- rowMeans(expr_tpm[,-1])\n",
    "print(sum(mean_expr >= quantile(mean_expr, prob)))\n",
    "expr_tpm_filtered <- expr_tpm[mean_expr >= prob,]\n",
    "\n",
    "# Log transform data\n",
    "# expr_tpm_log2 <- log2(expr_tpm_filtered[,-1] + 1)\n",
    "\n",
    "expr <- data.frame(Gene=expr_tpm_filtered[,1], expr_tpm_filtered[,-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "5cb6e63c",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Order samples by covariates of interest\n",
    "\n",
    "sampleinfo <- read.csv(\"/mnt/lareaulab/reliscu/projects/NSF_GRFP/data/bulk/hahn_2023/cortex/hahn_2023_cortex_sampleinfo.csv\")\n",
    "sampleinfo <- sampleinfo %>%\n",
    "    arrange(age, sex_y)\n",
    "sampleinfo <- sampleinfo[sampleinfo$Run %in% colnames(expr),]\n",
    "    \n",
    "expr <- expr[, c(1, match(sampleinfo$Run, colnames(expr)[-1]) + 1)]\n",
    "all.equal(sampleinfo$Run, colnames(expr)[-1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "b9a7e29f",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "samplegroups <- as.factor(sampleinfo$age)\n",
    "merge.param <- 0.85"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "62388941",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "597"
      ],
      "text/latex": [
       "597"
      ],
      "text/markdown": [
       "597"
      ],
      "text/plain": [
       "[1] 597"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Cga'</li><li>'Gh'</li><li>'Gm11192'</li><li>'Gm11594'</li><li>'Gm12571'</li><li>'Gm12719'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Cga'\n",
       "\\item 'Gh'\n",
       "\\item 'Gm11192'\n",
       "\\item 'Gm11594'\n",
       "\\item 'Gm12571'\n",
       "\\item 'Gm12719'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Cga'\n",
       "2. 'Gh'\n",
       "3. 'Gm11192'\n",
       "4. 'Gm11594'\n",
       "5. 'Gm12571'\n",
       "6. 'Gm12719'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Cga\"     \"Gh\"      \"Gm11192\" \"Gm11594\" \"Gm12571\" \"Gm12719\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Encountering errors when calculating distance in FM; problem seems related to some genes having 0 MAD (median absolute deviation)\n",
    "\n",
    "# Filtering these in advance:\n",
    "\n",
    "X <- as.matrix(expr[,-1])\n",
    "mads <- apply(X, 1, mad, na.rm = TRUE)\n",
    "bad  <- which(!is.finite(mads) | mads == 0)\n",
    "\n",
    "length(bad)\n",
    "head(expr[,1][bad])\n",
    "\n",
    "mask <- mads > 0\n",
    "expr <- expr[mask,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "b962a984",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 4</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Gene</th><th scope=col>SRR21355373</th><th scope=col>SRR21355377</th><th scope=col>SRR21355364</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>3941</th><td>0610009E02Rik</td><td>2.200651</td><td>3.060461</td><td>2.407408</td></tr>\n",
       "\t<tr><th scope=row>34041</th><td>0610009L18Rik</td><td>6.536998</td><td>6.141230</td><td>7.392501</td></tr>\n",
       "\t<tr><th scope=row>32471</th><td>0610010K14Rik</td><td>2.547075</td><td>1.828318</td><td>2.717159</td></tr>\n",
       "\t<tr><th scope=row>17310</th><td>0610030E20Rik</td><td>8.497339</td><td>7.914483</td><td>9.126372</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 4\n",
       "\\begin{tabular}{r|llll}\n",
       "  & Gene & SRR21355373 & SRR21355377 & SRR21355364\\\\\n",
       "  & <chr> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t3941 & 0610009E02Rik & 2.200651 & 3.060461 & 2.407408\\\\\n",
       "\t34041 & 0610009L18Rik & 6.536998 & 6.141230 & 7.392501\\\\\n",
       "\t32471 & 0610010K14Rik & 2.547075 & 1.828318 & 2.717159\\\\\n",
       "\t17310 & 0610030E20Rik & 8.497339 & 7.914483 & 9.126372\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 4\n",
       "\n",
       "| <!--/--> | Gene &lt;chr&gt; | SRR21355373 &lt;dbl&gt; | SRR21355377 &lt;dbl&gt; | SRR21355364 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|\n",
       "| 3941 | 0610009E02Rik | 2.200651 | 3.060461 | 2.407408 |\n",
       "| 34041 | 0610009L18Rik | 6.536998 | 6.141230 | 7.392501 |\n",
       "| 32471 | 0610010K14Rik | 2.547075 | 1.828318 | 2.717159 |\n",
       "| 17310 | 0610030E20Rik | 8.497339 | 7.914483 | 9.126372 |\n",
       "\n"
      ],
      "text/plain": [
       "      Gene          SRR21355373 SRR21355377 SRR21355364\n",
       "3941  0610009E02Rik 2.200651    3.060461    2.407408   \n",
       "34041 0610009L18Rik 6.536998    6.141230    7.392501   \n",
       "32471 0610010K14Rik 2.547075    1.828318    2.717159   \n",
       "17310 0610030E20Rik 8.497339    7.914483    9.126372   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "expr[1:4,1:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2727325b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "FindModules(\n",
    "  projectname=\"hahn_2023_cortex_STAR_TPM_OR_mergeParam0.85\",\n",
    "  expr=expr,\n",
    "  geneinfo=1,\n",
    "  sampleindex=2:ncol(expr),\n",
    "  samplegroups=samplegroups,\n",
    "  subset=NULL,\n",
    "  simMat=NULL,\n",
    "  saveSimMat=FALSE,\n",
    "  simType=\"Bicor\",\n",
    "  beta=1,\n",
    "  overlapType=\"None\",\n",
    "  TOtype=\"signed\",\n",
    "  TOdenom=\"min\",\n",
    "  MIestimator=\"mi.mm\",\n",
    "  MIdisc=\"equalfreq\",\n",
    "  signumType=\"rel\",\n",
    "  iterate=TRUE,\n",
    "  signumvec=c(.999,.99,.98,.97,.96,.85), # signumvec=c(.99999, .9999, .999, .99, .98, .97, .96, .95),\n",
    "  minsizevec=c(4,6,8,10,12,15), # minsizevec=c(3,4,5,6,8,10),\n",
    "  signum=NULL,\n",
    "  minSize=NULL,\n",
    "  merge.by=\"ME\",\n",
    "  merge.param=merge.param,\n",
    "  export.merge.comp=T,\n",
    "  ZNCcut=2,\n",
    "  calcSW=FALSE,\n",
    "  loadTree=FALSE,\n",
    "  writeKME=TRUE,\n",
    "  calcBigModStat=FALSE,\n",
    "  writeModSnap=TRUE\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e3251c1",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd44da6b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07087f85",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37a91fab",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3cd94ddc",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Debugging\n",
    "\n",
    "#   projectname=\"hahn_2023_cortex_STAR_TPM_SN_QN_mergeParam0.85\"\n",
    "#   expr=expr\n",
    "#   geneinfo=1\n",
    "#   sampleindex=2:ncol(expr)\n",
    "#   samplegroups=samplegroups\n",
    "#   subset=NULL\n",
    "#   simMat=NULL\n",
    "#   saveSimMat=FALSE\n",
    "#   simType=\"Bicor\"\n",
    "#   beta=1\n",
    "#   overlapType=\"None\"\n",
    "#   TOtype=\"signed\"\n",
    "#   TOdenom=\"min\"\n",
    "#   MIestimator=\"mi.mm\"\n",
    "#   MIdisc=\"equalfreq\"\n",
    "#   signumType=\"rel\"\n",
    "#   iterate=TRUE\n",
    "#   signumvec=c(.999, .99, .98, .97, .96, .85) # signumvec=c(.99999 .9999 .999 .99 .98 .97 .96 .95)\n",
    "#   minsizevec=c(4, 6, 8, 10, 12, 15) # minsizevec=c(3456810)\n",
    "#   signum=NULL\n",
    "#   minSize=NULL\n",
    "#   merge.by=\"ME\"\n",
    "#   merge.param=merge.param\n",
    "#   export.merge.comp=T\n",
    "#   ZNCcut=2\n",
    "#   calcSW=FALSE\n",
    "#   loadTree=FALSE\n",
    "#   writeKME=TRUE\n",
    "#   calcBigModStat=FALSE\n",
    "#   writeModSnap=TRUE\n",
    "#   varCut=NULL\n",
    "\n",
    "# \tif(length(unique(expr[,1]))!=length(expr[,1])){\n",
    "\t\t\n",
    "# \t\tstop(\"Identifiers in column 1 must be unique!\")\n",
    "\t\t\n",
    "# \t}\n",
    "\t\n",
    "# ## Create root directory, if necessary:\n",
    "\t\n",
    "# \tif(length(grep(paste(projectname,\"_Modules\",sep=\"\"),getwd()))==1){\n",
    "\t\t\n",
    "# \t\tbreakdir=strsplit(getwd(),split=\"/\")\n",
    "# \t\tBNroot=c(1:length(breakdir[[1]]))[is.element(breakdir[[1]],paste(projectname,\"_Modules\",sep=\"\"))]\n",
    "# \t\tsetwd(paste(breakdir[[1]][1:BNroot],collapse=\"/\"))\n",
    "\t\t\n",
    "# \t} else {\n",
    "\t\t\n",
    "# \t\tdir.create(paste(projectname,\"_Modules\",sep=\"\"))\n",
    "# \t\tsetwd(paste(getwd(),\"/\",projectname,\"_Modules\",sep=\"\"))\n",
    "\t\t\n",
    "# \t}\n",
    "\t\n",
    "# \tBNrootDir=getwd()\t\n",
    "\n",
    "# ## Check for missing data and set NA = 0 per user (recommended for RNAseq data):\n",
    "\n",
    "# \tmissing.values=sum(is.na(expr[,sampleindex]))\n",
    "# \ttotal.values=nrow(expr)*length(sampleindex)\n",
    "\t\n",
    "# \tif(missing.values>0){\n",
    "# \t  cat(\"\\n\")\n",
    "# \t\tprint(paste(signif(missing.values/total.values*100,2),\"% of data are missing. Setting NA = 0 (recommended for RNAseq data)\",sep=\"\"),quote=F)\n",
    "# \t\tcat(\"\\n\")\n",
    "# \t\texpr[,sampleindex][is.na(expr[,sampleindex])]=0\n",
    "# \t}\n",
    "\t\n",
    "# \t## Check for values <= 0, and scale so minimum values = 1 (avoids issues with log tranformations)\n",
    "\t\n",
    "# \tzero.values=sum(expr[,sampleindex]<=0)\n",
    "\t\n",
    "# \tif(zero.values>0){\n",
    "\t  \n",
    "# \t  cat(\"\\n\")\n",
    "# \t  print(\"Gene expression values <= 0 are present. Scaling all data so minimum expression = 1\",quote=F)\n",
    "# \t  cat(\"\\n\")\n",
    "# \t  expr[,sampleindex]=expr[,sampleindex]+abs(min(expr[,sampleindex],na.rm=T))+1\n",
    "\t  \n",
    "# \t} \n",
    "\t\n",
    "# ## Note: FM 0.88 excludes genes with low variance from kME table:\n",
    "\n",
    "# \tif(is.null(varCut)){\n",
    "# \t  varCut=0\n",
    "# \t}\n",
    "\t\n",
    "# \tdatExprall=t(expr[,sampleindex])\n",
    "# \tcolnames(datExprall)=as.character(expr[,1])\n",
    "# \tvar.all=apply(datExprall,2,var)\n",
    "# \tdatExprall=datExprall[,var.all>varCut] \n",
    "\t\n",
    "# \tif(sum(var.all<varCut)>0){\n",
    "\t\t\n",
    "# \t\tcat(\"\\n\")\n",
    "# \t\tprint(paste(\"Note: \",sum(var.all<varCut),\" features had 0 variance and were excluded\",sep=\"\"))\n",
    "# \t\tcat(\"\\n\")\n",
    "# \t\tExcludedFeatures=expr[var.all<varCut,geneinfo]\n",
    "# \t\twrite.table(ExcludedFeatures,file=\"Excluded_genes_with_0_variance.csv\",sep=\",\",row.names=F,col.names=T)\n",
    "\t\t\n",
    "# \t}\n",
    "\t\t\n",
    "# \tif(!is.null(subset)){\n",
    "\t\t\n",
    "# \t\tsubset=as.logical(subset)\n",
    "# \t\tdatExpr1=t(expr[subset,sampleindex])\n",
    "# \t\tcolnames(datExpr1)=as.character(expr[subset,1])\n",
    "\t\t\n",
    "# \t} else {\n",
    "\t\t\n",
    "# \t\tdatExpr1=t(expr[,sampleindex])\n",
    "# \t\tcolnames(datExpr1)=as.character(expr[,1])\n",
    "\t\t\n",
    "# \t}\n",
    "\t\n",
    "# \tvar1=apply(datExpr1,2,var)\n",
    "\t\n",
    "# \tif(sum(var1<varCut)>0){\n",
    "\t\t\n",
    "# \t\tdatExpr1=datExpr1[,var1>varCut]\n",
    "\t\t\n",
    "# \t}\n",
    "\t\n",
    "# \tif(is.null(samplegroups)){\n",
    "\t\t\n",
    "# \t\tsamplegroups=c(rep(1,length(sampleindex)))\n",
    "\t\t\n",
    "# \t} else {\n",
    "\t  \n",
    "# \t  if(sum(is.na(samplegroups))>0){\n",
    "\t    \n",
    "# \t    samplegroups=as.character(samplegroups)\n",
    "# \t    samplegroups[is.na(samplegroups)]=\"None\"\n",
    "\t    \n",
    "# \t  }\n",
    "\t\t\n",
    "# \t\tif(is.character(samplegroups)){\n",
    "\t\t\t\n",
    "# \t\t  samplegroups <- as.factor(samplegroups)\n",
    "\t\t\t\n",
    "# \t\t}\n",
    "\t\t\n",
    "# \t}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7376b18",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in bicor(datExpr1, use = \"p\"):\n",
      "“bicor: zero MAD in variable 'x'. Pearson correlation was used for individual columns with zero (or missing) MAD.”\n"
     ]
    }
   ],
   "source": [
    "# simMat=bicor(datExpr1,use=\"p\")\n",
    "# collectGarbage()\n",
    "# diag(simMat)=0\n",
    "# collectGarbage()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0d2479b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Testing parameters...\"\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in quantile.default(sample(simMat, 1e+06, replace = FALSE), probs = signumvec): missing values and NaN's not allowed if 'na.rm' is FALSE\n",
     "output_type": "error",
     "traceback": [
      "Error in quantile.default(sample(simMat, 1e+06, replace = FALSE), probs = signumvec): missing values and NaN's not allowed if 'na.rm' is FALSE\nTraceback:\n",
      "1. quantile.default(sample(simMat, 1e+06, replace = FALSE), probs = signumvec)",
      "2. stop(\"missing values and NaN's not allowed if 'na.rm' is FALSE\")",
      "3. .handleSimpleError(function (cnd) \n . {\n .     watcher$capture_plot_and_output()\n .     cnd <- sanitize_call(cnd)\n .     watcher$push(cnd)\n .     switch(on_error, continue = invokeRestart(\"eval_continue\"), \n .         stop = invokeRestart(\"eval_stop\"), error = NULL)\n . }, \"missing values and NaN's not allowed if 'na.rm' is FALSE\", \n .     base::quote(quantile.default(sample(simMat, 1e+06, replace = FALSE), \n .         probs = signumvec)))"
     ]
    }
   ],
   "source": [
    "# print(\"Testing parameters...\")\n",
    "# \t\tcat(\"\\n\")\n",
    "\t\t\n",
    "# \t\tif(signumType==\"rel\"){\n",
    "\t\t\n",
    "# \t\t\tif(((dim(datExpr1)[2]*(dim(datExpr1)[2]-1))/2)>1e+06){\n",
    "\t\t\t\n",
    "# \t\t\t\trelsignumvec=quantile(sample(simMat,1000000,replace=FALSE),probs=signumvec)\n",
    "# \t\t\t\tcollectGarbage()\n",
    "\t\t\t\n",
    "# \t\t\t} else {\n",
    "\t\t\n",
    "# \t\t\t\trelsignumvec=quantile(simMat[upper.tri(simMat)],probs=signumvec)\n",
    "# \t\t\t\tcollectGarbage()\n",
    "\t\t\t\n",
    "# \t\t\t}\n",
    "\t\t\t\n",
    "# \t\t}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9243eb32",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
