{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "df4f6ab5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "sys.path.append(\"/mnt/lareaulab/reliscu/code\")\n",
    "\n",
    "from parse_gtf import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cc2be6f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "psi = pd.read_csv(f\"data/hahn_2023_cortex_STAR_exon_PSI.csv\", index_col=0)\n",
    "top_qval_mods_df = pd.read_csv(\"data/enrichments/hahn_2023_cortex_STAR_TPM_OR_QN_mergeParam0.85_top_MO_Qval_modules.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c96847b3",
   "metadata": {},
   "source": [
    "### Add gene names to PSI data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f11ee42a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parse GTF attribute column\n",
    "gtf_file = \"/mnt/lareaulab/reliscu/data/GENCODE/GRCm39/gencode.vM35.annotation.gtf\"\n",
    "gtf = gtf_parse(gtf_file)\n",
    "gtf_subset = gtf.loc[gtf['feature'].isin([\"gene\"])]\n",
    "attrs = gtf_subset[\"attribute\"].apply(extract_attributes)\n",
    "attrs_df = attrs.apply(pd.Series)\n",
    "gtf_parsed = pd.concat([gtf_subset.drop(columns=[\"attribute\"]), attrs_df], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "054bd5ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get PSI and GTF data ready to merge on gene IDs\n",
    "gtf_parsed['gene_id'] = gtf_parsed['gene_id'].str.split(\".\").str[0]\n",
    "psi['gene_id'] = psi.index.str.split(\"_\").str[0]\n",
    "psi['exon_id'] = psi.index.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b4c7976d",
   "metadata": {},
   "outputs": [],
   "source": [
    "psi_anno = pd.merge(gtf_parsed[['gene_id', 'gene_name']], psi, on=\"gene_id\", how=\"right\")\n",
    "psi_anno = psi_anno.set_index(\"exon_id\").rename_axis(None)\n",
    "psi_anno = psi_anno.drop(columns=[\"gene_id\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80e4980c",
   "metadata": {},
   "source": [
    "### Calc. corr between ME and exon PSI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "87933e22",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2897: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[:, None]\n",
      "/mnt/lareaulab/reliscu/anaconda3/envs/anndata/lib/python3.12/site-packages/numpy/lib/function_base.py:2898: RuntimeWarning: invalid value encountered in divide\n",
      "  c /= stddev[None, :]\n"
     ]
    }
   ],
   "source": [
    "corr_df = pd.DataFrame(\n",
    "    columns=[\"Gene\"] + top_qval_mods_df['Cell_type'].tolist(), \n",
    "    index=psi_anno.index\n",
    ")\n",
    "corr_df['Gene'] = psi_anno['gene_name'] \n",
    "\n",
    "for i, row in top_qval_mods_df.iterrows():\n",
    "    ctype = row['Cell_type']\n",
    "\n",
    "    mod_df = pd.read_csv(row['ME_path'])\n",
    "    mod_eig = mod_df.set_index(\"Sample\")[row['Module']]\n",
    "    mod_eig = pd.to_numeric(mod_eig, errors=\"coerce\")\n",
    "    \n",
    "    corrs = psi_anno.iloc[:, 1:].corrwith(mod_eig, axis=1)\n",
    "    corr_df[ctype] = corrs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e3b9b53c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>YANG_PFC_2021_OLIGODENDROCYTE</th>\n",
       "      <th>BARRES_OLIGODENDROCYTES</th>\n",
       "      <th>YANG_PFC_2021_NRGN_NEURON</th>\n",
       "      <th>BAKKEN_2019_PVALB_GABAERGIC_DE_GABA_CLUSTERS</th>\n",
       "      <th>YANG_PFC_2021_L2_3_EXCITATORY_NEURON</th>\n",
       "      <th>BAKKEN_2019_SST_CHODL_GABAERGIC_DE_GABA_CLUSTERS</th>\n",
       "      <th>YANG_PFC_2021_ASTROCYTE</th>\n",
       "      <th>Mukamel_ExcitatoryNeurons</th>\n",
       "      <th>YANG_PFC_2021_OPC</th>\n",
       "      <th>...</th>\n",
       "      <th>SUGINO_UP_LAYERS4-6_GABAERGIC_NEURONS_CINGULATE_CTX</th>\n",
       "      <th>SCHIRMER_2019_OPC</th>\n",
       "      <th>Mukamel_InhibitoryNeurons</th>\n",
       "      <th>ZEISEL_MURAL</th>\n",
       "      <th>SCHIRMER_2019_MICROGLIA</th>\n",
       "      <th>SCHIRMER_2019_IN_VIP</th>\n",
       "      <th>SCHIRMER_2019_IN_PVALB</th>\n",
       "      <th>SCHIRMER_2019_IN_SV2C</th>\n",
       "      <th>SCHIRMER_2019_IN_SST</th>\n",
       "      <th>SCHIRMER_2019_EX_PYR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000025902_ProteinCoding_1</th>\n",
       "      <td>Sox17</td>\n",
       "      <td>0.092357</td>\n",
       "      <td>0.095700</td>\n",
       "      <td>0.131528</td>\n",
       "      <td>0.130839</td>\n",
       "      <td>0.124370</td>\n",
       "      <td>-0.251925</td>\n",
       "      <td>0.065036</td>\n",
       "      <td>0.128046</td>\n",
       "      <td>-0.101613</td>\n",
       "      <td>...</td>\n",
       "      <td>0.138981</td>\n",
       "      <td>-0.211470</td>\n",
       "      <td>-0.009544</td>\n",
       "      <td>0.082746</td>\n",
       "      <td>0.131194</td>\n",
       "      <td>0.096587</td>\n",
       "      <td>-0.150152</td>\n",
       "      <td>-0.170045</td>\n",
       "      <td>0.213244</td>\n",
       "      <td>-0.130622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000025902_ProteinCoding_2</th>\n",
       "      <td>Sox17</td>\n",
       "      <td>0.079432</td>\n",
       "      <td>0.083039</td>\n",
       "      <td>0.127964</td>\n",
       "      <td>0.128391</td>\n",
       "      <td>0.122055</td>\n",
       "      <td>-0.260303</td>\n",
       "      <td>0.069982</td>\n",
       "      <td>0.125595</td>\n",
       "      <td>-0.101295</td>\n",
       "      <td>...</td>\n",
       "      <td>0.137899</td>\n",
       "      <td>-0.196701</td>\n",
       "      <td>-0.000914</td>\n",
       "      <td>0.085217</td>\n",
       "      <td>0.128936</td>\n",
       "      <td>0.083223</td>\n",
       "      <td>-0.140479</td>\n",
       "      <td>-0.167023</td>\n",
       "      <td>0.206923</td>\n",
       "      <td>-0.142831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000025902_ProteinCoding_3</th>\n",
       "      <td>Sox17</td>\n",
       "      <td>-0.043981</td>\n",
       "      <td>-0.042342</td>\n",
       "      <td>0.121537</td>\n",
       "      <td>0.119390</td>\n",
       "      <td>0.119838</td>\n",
       "      <td>-0.122498</td>\n",
       "      <td>-0.096126</td>\n",
       "      <td>0.115919</td>\n",
       "      <td>-0.107698</td>\n",
       "      <td>...</td>\n",
       "      <td>0.133972</td>\n",
       "      <td>-0.094439</td>\n",
       "      <td>0.162286</td>\n",
       "      <td>-0.101081</td>\n",
       "      <td>0.119646</td>\n",
       "      <td>-0.046986</td>\n",
       "      <td>-0.094943</td>\n",
       "      <td>-0.072828</td>\n",
       "      <td>0.168193</td>\n",
       "      <td>-0.151309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000025902_ProteinCoding_4</th>\n",
       "      <td>Sox17</td>\n",
       "      <td>-0.023505</td>\n",
       "      <td>-0.023307</td>\n",
       "      <td>0.085887</td>\n",
       "      <td>0.086728</td>\n",
       "      <td>0.086286</td>\n",
       "      <td>-0.091748</td>\n",
       "      <td>-0.121977</td>\n",
       "      <td>0.084180</td>\n",
       "      <td>-0.064705</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102204</td>\n",
       "      <td>-0.017887</td>\n",
       "      <td>0.109462</td>\n",
       "      <td>-0.129700</td>\n",
       "      <td>0.087563</td>\n",
       "      <td>-0.025678</td>\n",
       "      <td>-0.137717</td>\n",
       "      <td>-0.127460</td>\n",
       "      <td>0.090053</td>\n",
       "      <td>-0.097208</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000033845_ProteinCoding_1</th>\n",
       "      <td>Mrpl15</td>\n",
       "      <td>-0.046297</td>\n",
       "      <td>-0.049318</td>\n",
       "      <td>-0.222226</td>\n",
       "      <td>-0.225703</td>\n",
       "      <td>-0.226101</td>\n",
       "      <td>0.137570</td>\n",
       "      <td>0.142890</td>\n",
       "      <td>-0.227870</td>\n",
       "      <td>0.220931</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.220301</td>\n",
       "      <td>0.126290</td>\n",
       "      <td>-0.203138</td>\n",
       "      <td>0.186141</td>\n",
       "      <td>-0.226582</td>\n",
       "      <td>-0.044729</td>\n",
       "      <td>-0.024642</td>\n",
       "      <td>-0.136517</td>\n",
       "      <td>-0.136999</td>\n",
       "      <td>0.202147</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene  YANG_PFC_2021_OLIGODENDROCYTE  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1   Sox17                       0.092357   \n",
       "ENSMUSG00000025902_ProteinCoding_2   Sox17                       0.079432   \n",
       "ENSMUSG00000025902_ProteinCoding_3   Sox17                      -0.043981   \n",
       "ENSMUSG00000025902_ProteinCoding_4   Sox17                      -0.023505   \n",
       "ENSMUSG00000033845_ProteinCoding_1  Mrpl15                      -0.046297   \n",
       "\n",
       "                                    BARRES_OLIGODENDROCYTES  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                 0.095700   \n",
       "ENSMUSG00000025902_ProteinCoding_2                 0.083039   \n",
       "ENSMUSG00000025902_ProteinCoding_3                -0.042342   \n",
       "ENSMUSG00000025902_ProteinCoding_4                -0.023307   \n",
       "ENSMUSG00000033845_ProteinCoding_1                -0.049318   \n",
       "\n",
       "                                    YANG_PFC_2021_NRGN_NEURON  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                   0.131528   \n",
       "ENSMUSG00000025902_ProteinCoding_2                   0.127964   \n",
       "ENSMUSG00000025902_ProteinCoding_3                   0.121537   \n",
       "ENSMUSG00000025902_ProteinCoding_4                   0.085887   \n",
       "ENSMUSG00000033845_ProteinCoding_1                  -0.222226   \n",
       "\n",
       "                                    BAKKEN_2019_PVALB_GABAERGIC_DE_GABA_CLUSTERS  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                                      0.130839   \n",
       "ENSMUSG00000025902_ProteinCoding_2                                      0.128391   \n",
       "ENSMUSG00000025902_ProteinCoding_3                                      0.119390   \n",
       "ENSMUSG00000025902_ProteinCoding_4                                      0.086728   \n",
       "ENSMUSG00000033845_ProteinCoding_1                                     -0.225703   \n",
       "\n",
       "                                    YANG_PFC_2021_L2_3_EXCITATORY_NEURON  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                              0.124370   \n",
       "ENSMUSG00000025902_ProteinCoding_2                              0.122055   \n",
       "ENSMUSG00000025902_ProteinCoding_3                              0.119838   \n",
       "ENSMUSG00000025902_ProteinCoding_4                              0.086286   \n",
       "ENSMUSG00000033845_ProteinCoding_1                             -0.226101   \n",
       "\n",
       "                                    BAKKEN_2019_SST_CHODL_GABAERGIC_DE_GABA_CLUSTERS  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                                         -0.251925   \n",
       "ENSMUSG00000025902_ProteinCoding_2                                         -0.260303   \n",
       "ENSMUSG00000025902_ProteinCoding_3                                         -0.122498   \n",
       "ENSMUSG00000025902_ProteinCoding_4                                         -0.091748   \n",
       "ENSMUSG00000033845_ProteinCoding_1                                          0.137570   \n",
       "\n",
       "                                    YANG_PFC_2021_ASTROCYTE  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                 0.065036   \n",
       "ENSMUSG00000025902_ProteinCoding_2                 0.069982   \n",
       "ENSMUSG00000025902_ProteinCoding_3                -0.096126   \n",
       "ENSMUSG00000025902_ProteinCoding_4                -0.121977   \n",
       "ENSMUSG00000033845_ProteinCoding_1                 0.142890   \n",
       "\n",
       "                                    Mukamel_ExcitatoryNeurons  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                   0.128046   \n",
       "ENSMUSG00000025902_ProteinCoding_2                   0.125595   \n",
       "ENSMUSG00000025902_ProteinCoding_3                   0.115919   \n",
       "ENSMUSG00000025902_ProteinCoding_4                   0.084180   \n",
       "ENSMUSG00000033845_ProteinCoding_1                  -0.227870   \n",
       "\n",
       "                                    YANG_PFC_2021_OPC  ...  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1          -0.101613  ...   \n",
       "ENSMUSG00000025902_ProteinCoding_2          -0.101295  ...   \n",
       "ENSMUSG00000025902_ProteinCoding_3          -0.107698  ...   \n",
       "ENSMUSG00000025902_ProteinCoding_4          -0.064705  ...   \n",
       "ENSMUSG00000033845_ProteinCoding_1           0.220931  ...   \n",
       "\n",
       "                                    SUGINO_UP_LAYERS4-6_GABAERGIC_NEURONS_CINGULATE_CTX  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                                           0.138981     \n",
       "ENSMUSG00000025902_ProteinCoding_2                                           0.137899     \n",
       "ENSMUSG00000025902_ProteinCoding_3                                           0.133972     \n",
       "ENSMUSG00000025902_ProteinCoding_4                                           0.102204     \n",
       "ENSMUSG00000033845_ProteinCoding_1                                          -0.220301     \n",
       "\n",
       "                                    SCHIRMER_2019_OPC  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1          -0.211470   \n",
       "ENSMUSG00000025902_ProteinCoding_2          -0.196701   \n",
       "ENSMUSG00000025902_ProteinCoding_3          -0.094439   \n",
       "ENSMUSG00000025902_ProteinCoding_4          -0.017887   \n",
       "ENSMUSG00000033845_ProteinCoding_1           0.126290   \n",
       "\n",
       "                                    Mukamel_InhibitoryNeurons  ZEISEL_MURAL  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                  -0.009544      0.082746   \n",
       "ENSMUSG00000025902_ProteinCoding_2                  -0.000914      0.085217   \n",
       "ENSMUSG00000025902_ProteinCoding_3                   0.162286     -0.101081   \n",
       "ENSMUSG00000025902_ProteinCoding_4                   0.109462     -0.129700   \n",
       "ENSMUSG00000033845_ProteinCoding_1                  -0.203138      0.186141   \n",
       "\n",
       "                                    SCHIRMER_2019_MICROGLIA  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1                 0.131194   \n",
       "ENSMUSG00000025902_ProteinCoding_2                 0.128936   \n",
       "ENSMUSG00000025902_ProteinCoding_3                 0.119646   \n",
       "ENSMUSG00000025902_ProteinCoding_4                 0.087563   \n",
       "ENSMUSG00000033845_ProteinCoding_1                -0.226582   \n",
       "\n",
       "                                    SCHIRMER_2019_IN_VIP  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1              0.096587   \n",
       "ENSMUSG00000025902_ProteinCoding_2              0.083223   \n",
       "ENSMUSG00000025902_ProteinCoding_3             -0.046986   \n",
       "ENSMUSG00000025902_ProteinCoding_4             -0.025678   \n",
       "ENSMUSG00000033845_ProteinCoding_1             -0.044729   \n",
       "\n",
       "                                    SCHIRMER_2019_IN_PVALB  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1               -0.150152   \n",
       "ENSMUSG00000025902_ProteinCoding_2               -0.140479   \n",
       "ENSMUSG00000025902_ProteinCoding_3               -0.094943   \n",
       "ENSMUSG00000025902_ProteinCoding_4               -0.137717   \n",
       "ENSMUSG00000033845_ProteinCoding_1               -0.024642   \n",
       "\n",
       "                                    SCHIRMER_2019_IN_SV2C  \\\n",
       "ENSMUSG00000025902_ProteinCoding_1              -0.170045   \n",
       "ENSMUSG00000025902_ProteinCoding_2              -0.167023   \n",
       "ENSMUSG00000025902_ProteinCoding_3              -0.072828   \n",
       "ENSMUSG00000025902_ProteinCoding_4              -0.127460   \n",
       "ENSMUSG00000033845_ProteinCoding_1              -0.136517   \n",
       "\n",
       "                                    SCHIRMER_2019_IN_SST  SCHIRMER_2019_EX_PYR  \n",
       "ENSMUSG00000025902_ProteinCoding_1              0.213244             -0.130622  \n",
       "ENSMUSG00000025902_ProteinCoding_2              0.206923             -0.142831  \n",
       "ENSMUSG00000025902_ProteinCoding_3              0.168193             -0.151309  \n",
       "ENSMUSG00000025902_ProteinCoding_4              0.090053             -0.097208  \n",
       "ENSMUSG00000033845_ProteinCoding_1             -0.136999              0.202147  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "3805f50d",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df.to_csv(f\"data/corrs/hahn_2023_cortex_STAR_TPM_OR_QN_mergeParam0.85_top_MO_Qval_modules_exon_corr.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "anndata",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
