{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "df4f6ab5",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "sys.path.append(\"/mnt/lareaulab/reliscu/code\")\n",
    "\n",
    "from parse_gtf import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc2be6f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "psi = pd.read_csv(f\"data/hahn_2023_cortex_STAR_exon_PSI.csv\")\n",
    "top_qval_mods_df = pd.read_csv(\"data/enrichments/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c96847b3",
   "metadata": {},
   "source": [
    "### Add gene names to PSI data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f11ee42a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parse GTF attribute column\n",
    "gtf_file = \"/mnt/lareaulab/reliscu/data/GENCODE/GRCm39/gencode.vM35.annotation.gtf\"\n",
    "gtf = gtf_parse(gtf_file)\n",
    "gtf_subset = gtf.loc[gtf['feature'].isin([\"gene\"])]\n",
    "attrs = gtf_subset[\"attribute\"].apply(extract_attributes)\n",
    "attrs_df = attrs.apply(pd.Series)\n",
    "gtf_parsed = pd.concat([gtf_subset.drop(columns=[\"attribute\"]), attrs_df], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "054bd5ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get PSI and GTF data ready to merge on gene IDs\n",
    "gtf_parsed['gene_id'] = gtf_parsed['gene_id'].str.split(\".\").str[0]\n",
    "psi['gene_id'] = psi.index.str.split(\"_\").str[0]\n",
    "psi['exon_id'] = psi.index.values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "b4c7976d",
   "metadata": {},
   "outputs": [],
   "source": [
    "psi_anno = pd.merge(gtf_parsed[['gene_id', 'gene_name']], psi, on=\"gene_id\", how=\"right\")\n",
    "psi_anno = psi_anno.set_index(\"exon_id\").rename_axis(None)\n",
    "psi_anno = psi_anno.drop(columns=[\"gene_id\"])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "80e4980c",
   "metadata": {},
   "source": [
    "### Calc. corr between ME and exon PSI"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "87933e22",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df = pd.DataFrame(\n",
    "    columns=[\"Gene\"] + top_qval_mods_df['Cell_type'].tolist(), \n",
    "    index=psi_anno.index\n",
    ")\n",
    "corr_df['Gene'] = psi_anno['gene_name'] \n",
    "\n",
    "for i, row in top_qval_mods_df.iterrows():\n",
    "    ctype = row['Cell_type']\n",
    "\n",
    "    mod_df = pd.read_csv(row['ME_path'])\n",
    "    mod_eig = mod_df.set_index(\"Sample\")[row['Module']]\n",
    "    mod_eig = pd.to_numeric(mod_eig, errors=\"coerce\")\n",
    "    \n",
    "    corrs = psi_anno.iloc[:, 1:].corrwith(mod_eig, axis=1)\n",
    "    corr_df[ctype] = corrs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e3b9b53c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>SMC-Peri</th>\n",
       "      <th>Pvalb</th>\n",
       "      <th>Lamp5</th>\n",
       "      <th>L2_3_IT</th>\n",
       "      <th>L6b</th>\n",
       "      <th>L6_CT</th>\n",
       "      <th>L4_5_IT</th>\n",
       "      <th>Vip</th>\n",
       "      <th>L5_6_NP</th>\n",
       "      <th>Sst_Chodl</th>\n",
       "      <th>L6_IT</th>\n",
       "      <th>Sncg</th>\n",
       "      <th>L5_PT</th>\n",
       "      <th>L5_IT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000033845_ProteinCoding_1</th>\n",
       "      <td>Mrpl15</td>\n",
       "      <td>0.292223</td>\n",
       "      <td>-0.144820</td>\n",
       "      <td>0.020312</td>\n",
       "      <td>0.186401</td>\n",
       "      <td>0.018275</td>\n",
       "      <td>-0.197350</td>\n",
       "      <td>0.037798</td>\n",
       "      <td>-0.071198</td>\n",
       "      <td>-0.026732</td>\n",
       "      <td>-0.015332</td>\n",
       "      <td>0.093463</td>\n",
       "      <td>-0.002950</td>\n",
       "      <td>0.128743</td>\n",
       "      <td>-0.138159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000025903_ProteinCoding_1</th>\n",
       "      <td>Lypla1</td>\n",
       "      <td>0.010090</td>\n",
       "      <td>-0.047604</td>\n",
       "      <td>-0.153901</td>\n",
       "      <td>0.209876</td>\n",
       "      <td>-0.062760</td>\n",
       "      <td>-0.020802</td>\n",
       "      <td>0.043826</td>\n",
       "      <td>0.130093</td>\n",
       "      <td>0.039971</td>\n",
       "      <td>0.103549</td>\n",
       "      <td>-0.023690</td>\n",
       "      <td>-0.060934</td>\n",
       "      <td>0.106918</td>\n",
       "      <td>-0.158337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000025903_ProteinCoding_2</th>\n",
       "      <td>Lypla1</td>\n",
       "      <td>0.087887</td>\n",
       "      <td>-0.055226</td>\n",
       "      <td>0.058410</td>\n",
       "      <td>0.134998</td>\n",
       "      <td>-0.174957</td>\n",
       "      <td>-0.066532</td>\n",
       "      <td>0.017559</td>\n",
       "      <td>0.035040</td>\n",
       "      <td>0.077610</td>\n",
       "      <td>0.050214</td>\n",
       "      <td>-0.049983</td>\n",
       "      <td>-0.008342</td>\n",
       "      <td>0.159381</td>\n",
       "      <td>-0.132973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000002459_ProteinCoding_1</th>\n",
       "      <td>Rgs20</td>\n",
       "      <td>0.044754</td>\n",
       "      <td>-0.123480</td>\n",
       "      <td>-0.132085</td>\n",
       "      <td>-0.088376</td>\n",
       "      <td>0.110440</td>\n",
       "      <td>0.096169</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>-0.001209</td>\n",
       "      <td>0.040877</td>\n",
       "      <td>-0.143903</td>\n",
       "      <td>0.123422</td>\n",
       "      <td>-0.076833</td>\n",
       "      <td>0.212663</td>\n",
       "      <td>0.103155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000002459_ProteinCoding_2</th>\n",
       "      <td>Rgs20</td>\n",
       "      <td>0.050632</td>\n",
       "      <td>0.104558</td>\n",
       "      <td>0.031212</td>\n",
       "      <td>0.028741</td>\n",
       "      <td>0.224542</td>\n",
       "      <td>0.020624</td>\n",
       "      <td>-0.123236</td>\n",
       "      <td>-0.071114</td>\n",
       "      <td>0.074500</td>\n",
       "      <td>0.071965</td>\n",
       "      <td>-0.051668</td>\n",
       "      <td>0.118239</td>\n",
       "      <td>-0.128358</td>\n",
       "      <td>-0.143223</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      Gene  SMC-Peri     Pvalb     Lamp5  \\\n",
       "ENSMUSG00000033845_ProteinCoding_1  Mrpl15  0.292223 -0.144820  0.020312   \n",
       "ENSMUSG00000025903_ProteinCoding_1  Lypla1  0.010090 -0.047604 -0.153901   \n",
       "ENSMUSG00000025903_ProteinCoding_2  Lypla1  0.087887 -0.055226  0.058410   \n",
       "ENSMUSG00000002459_ProteinCoding_1   Rgs20  0.044754 -0.123480 -0.132085   \n",
       "ENSMUSG00000002459_ProteinCoding_2   Rgs20  0.050632  0.104558  0.031212   \n",
       "\n",
       "                                     L2_3_IT       L6b     L6_CT   L4_5_IT  \\\n",
       "ENSMUSG00000033845_ProteinCoding_1  0.186401  0.018275 -0.197350  0.037798   \n",
       "ENSMUSG00000025903_ProteinCoding_1  0.209876 -0.062760 -0.020802  0.043826   \n",
       "ENSMUSG00000025903_ProteinCoding_2  0.134998 -0.174957 -0.066532  0.017559   \n",
       "ENSMUSG00000002459_ProteinCoding_1 -0.088376  0.110440  0.096169  0.000393   \n",
       "ENSMUSG00000002459_ProteinCoding_2  0.028741  0.224542  0.020624 -0.123236   \n",
       "\n",
       "                                         Vip   L5_6_NP  Sst_Chodl     L6_IT  \\\n",
       "ENSMUSG00000033845_ProteinCoding_1 -0.071198 -0.026732  -0.015332  0.093463   \n",
       "ENSMUSG00000025903_ProteinCoding_1  0.130093  0.039971   0.103549 -0.023690   \n",
       "ENSMUSG00000025903_ProteinCoding_2  0.035040  0.077610   0.050214 -0.049983   \n",
       "ENSMUSG00000002459_ProteinCoding_1 -0.001209  0.040877  -0.143903  0.123422   \n",
       "ENSMUSG00000002459_ProteinCoding_2 -0.071114  0.074500   0.071965 -0.051668   \n",
       "\n",
       "                                        Sncg     L5_PT     L5_IT  \n",
       "ENSMUSG00000033845_ProteinCoding_1 -0.002950  0.128743 -0.138159  \n",
       "ENSMUSG00000025903_ProteinCoding_1 -0.060934  0.106918 -0.158337  \n",
       "ENSMUSG00000025903_ProteinCoding_2 -0.008342  0.159381 -0.132973  \n",
       "ENSMUSG00000002459_ProteinCoding_1 -0.076833  0.212663  0.103155  \n",
       "ENSMUSG00000002459_ProteinCoding_2  0.118239 -0.128358 -0.143223  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3805f50d",
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df.to_csv(f\"data/hahn_2023_cortex_STAR_{psi_data}_exon_corr.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
